### PR TITLE
Allow casts from the result of `checked_abs` to unsigned

### LINKED ISSUE #4743: False positive for cast_sign_loss
## `cargo clippy -V`
```
clippy 0.0.212 (52cebb1 2019-10-15)
```
## Example code
```rust
#![warn(clippy::cast_sign_loss)]

fn main() {
    let value = i64::min_value().checked_abs().unwrap_or(0) as u64;
    
    assert_eq!(value, 0);
}
```
[Playground](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=81cf39e78646285ec19d8b42f2d2a96b)

## Problem
Clippy does warn, that the sign could be lost implicitly, even though it is done explicitly with [`checked_abs`](https://doc.rust-lang.org/std/primitive.i64.html#method.checked_abs), which either returns `Some(positive i64)` or `None`.

## Expected behavior
Clippy should ignore this case.

### PR DESCRIPTION
**Pull request by chansuke**
Fixes #4743.

### CODE REVIEW

File: tests/ui/cast.rs [hunk 1]
```diff
@@ -47,4 +47,9 @@ fn main() {
     (-1i32).abs() as u32;
     (-1i64).abs() as u64;
     (-1isize).abs() as usize;
+    (-1i8).checked_abs() as u8;
+    (-1i16).checked_abs() as u16;
+    (-1i32).checked_abs() as u32;
+    (-1i64).checked_abs() as u64;
+    (-1isize).checked_abs() as usize;
```
Comment (JohnTitor) [inline tests/ui/cast.rs]: `checked_abs` returns `Option<T>` so we should use `unwrap` or something else to get contained values.

Comment (chansuke) [reply tests/ui/cast.rs]: Thank you for the review, i will fix that

File: clippy_lints/src/types.rs [hunk 2]
```diff
@@ -1031,6 +1031,15 @@ fn check_loss_of_sign(cx: &LateContext<'_, '_>, expr: &Expr, op: &Expr, cast_fro
         }
     }
 
+    // don't lint for the result of `checked_abs`
+    if_chain! {
+        if let ExprKind::MethodCall(ref path, _, _) = op.kind;
+        if path.ident.name.as_str() == "checked_abs";
```
Comment (flip1995) [inline clippy_lints/src/types.rs]: Why not include this check in the if_chain above? 
```rust
        if ["abs", "checked_abs"].contains(path.ident.name);
```

**Comment by phansch** [comment]
triggering a new build

File: clippy_lints/src/types.rs (hunk 2 above)
Comment (chansuke) [reply clippy_lints/src/types.rs]: @flip1995 Thank you for the review. I have fixed the duplication.

**Comment by chansuke** [comment]
@flip1995 Thank you!!

File: clippy_lints/src/types.rs [hunk 3]
```diff
@@ -1020,12 +1020,12 @@ fn check_loss_of_sign(cx: &LateContext<'_, '_>, expr: &Expr, op: &Expr, cast_fro
...
     }
 
-    // don't lint for the result of `abs`
+    // don't lint for the result of `abs` & `checked_abs`
     // `abs` is an inherent impl of `i{N}`, so a method call with ident `abs` will always
     // resolve to that spesific method
     if_chain! {
         if let ExprKind::MethodCall(ref path, _, _) = op.kind;
-        if path.ident.name.as_str() == "abs";
+        if ["abs", "checked_abs"].contains(path.ident.name);
```
Comment (flip1995) [inline clippy_lints/src/types.rs]: `path.indent.name` gives a type error

**Comment by phansch** [comment]
It seems a fix for #4743 was already included in #4883, so I'm going ahead and close this PR. In any case, thank you for your time spent on improving Clippy :heart_decoration:

**Pull request closed without merging**